spaceinvaders.py

```diff
@@ -316,21 +316,32 @@ def reset(self, score, lives):
 		self.killedArray = [[0] * 10 for x in range(5)]
 
 	def make_blockers(self, number):
-	   blockerGroup = sprite.Group()
-	   for row in range(4):
-		   for column in range(9):
-			   blocker = Blocker(10, GREEN, row, column)
-			   blocker.rect.x = 50 + (200 * number) + (column * blocker.width)
-			   blocker.rect.y = 450 + (row * blocker.height)
-			   blockerGroup.add(blocker)
-	   return blockerGroup
-
-	def reset_lives(self):
+		blockerGroup = sprite.Group()
+		for row in range(4):
+			for column in range(9):
+				blocker = Blocker(10, GREEN, row, column)
+				blocker.rect.x = 50 + (200 * number) + (column * blocker.width)
+				blocker.rect.y = 450 + (row * blocker.height)
+				blockerGroup.add(blocker)
+		return blockerGroup
+	
+	def reset_lives_sprites(self):
 		self.life1 = Life(715, 3)
 		self.life2 = Life(742, 3)
 		self.life3 = Life(769, 3)
-		self.livesGroup = sprite.Group(self.life1, self.life2, self.life3)
 		
+		if self.lives == 3:
+			self.livesGroup = sprite.Group(self.life1, self.life2, self.life3)
+		elif self.lives == 2:
+			self.lives -= 1
```

Comment:
Is this supposed to be in here? Why -1 only on 2 lives?

Reply:
Good spottin, must have been a c+p error. The lives are decremented in check_collisions() so no reason to decrement here. Decrementing worked fine before, just the sprites were not handled correctly on a new round